### PR TITLE
HRTF optimizations

### PR DESCRIPTION
**Pull request by kencooke**
This PR improves Audio-Mixer scalability on the latest Intel CPUs (Core i9 and Skylake-SP Xeon). The framework is added to detect and use AVX512 instructions.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/981/HighFidelity-Beta-PR10780-38c87b0f0800cbe2917e87b98d2ed25152e5130c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/981/HighFidelity-Beta-PR10780-38c87b0f0800cbe2917e87b98d2ed25152e5130c.exe

**Comment by ZappoMan** [comment]
Cool!

**Comment by kencooke** [comment]
### Test Plan

The AVX512 code is hard to test until the Amazon EC2 C5 node becomes available, so it has been verified by unit-test on an emulator. The existing code is unchanged.

Please smoke-test the audio using both Interface and Sandbox from this build. Spatialized audio should work as expected.

**Comment by YuppyPlays** [comment]
Can test when built. Though I assume an i7 devils cayon will suffice, or does it need to be one of the specific ones mentioned above?

**Comment by kencooke** [comment]
@YuppyPlays thanks! i7 is fine. The others are not widely available yet.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/993/HighFidelity-Beta-PR10780-af0f4606592b2d2a03ab388fd833d9fbef85ba61.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/993/HighFidelity-Beta-PR10780-af0f4606592b2d2a03ab388fd833d9fbef85ba61.dmg

**Comment by YuppyPlays** [comment]
Testing complete:

Audio appears to work normally when listening the spacing seems normal (ie, things from the left project sound from the left set of speakers dominantly, and vise versa, center, distance, etc) all in Welcome & in hosted sandbox server.

**Pass**

File: libraries/shared/src/CPUDetect.h [hunk 1]
```diff
@@ -13,66 +13,106 @@
...
+#else
+
+static inline void cpuidex(int info[4], int eax, int ecx) {
+    info[0] = 0;
+    info[1] = 0;
+    info[2] = 0;
+    info[3] = 0;
+}
+
+static inline unsigned long long xgetbv(unsigned int ecx){
```
Comment (AndrewMeadows) [inline libraries/shared/src/CPUDetect.h]: Not a blocker: I find `uint64_t` more readable than `unsigned long long`.  Similarly `uint32_t` over `unsigned int`.  But if you were to adopt that pattern `int32_t` should be preferred over `int`.

Comment (kencooke) [inline libraries/shared/src/CPUDetect.h]: I agree. The reason I didn't use stdint.h types here was to better match the MSVC function I am emulating:
```
unsigned __int64 _xgetbv(unsigned int)
```

**Pull request merged**